На проверку

ZlobinGM opened the pull request.

ZlobinGM committed:
Ver 0.1 add stack

ZlobinGM committed:
Ver 0.1

ZlobinGM committed:
Rename

ZlobinGM committed:
Ver 0.11

ZlobinGM committed:
Ver 0.2 add new method to stack and transformation to normal form in postfix

ZlobinGM committed:
Ver 0.3

ZlobinGM committed:
Ver 0.7 google test launch problem

ZlobinGM committed:
Ver 1.0 google tests works (i dont include postfix.cpp in test_tpostfix.cpp)

alvls commented:
Георгий, с одной стороны с тестами и реализацией класса Постфикс все хорошо. С другой, программа-пример, повисает при вводе выражения "-a+b" или "-1+2" и падает при вводе "a+-b". Предлагаю что-то с этим сделать.

ZlobinGM committed:
не завершаемость исправлена, а вылет хоть и исправлен, но остался вопрос

Я решил ловить в main бросаемое исклчение, но если его не выловить, то в Злобин\base\postfix.cpp строка 113 return postfix; бросает исключение "Возникло необработанное исключение по адресу 0x75E53572 в base.exe: исключение Microsoft C++: char по адресу памяти 0x0060F8F4."
Вопрос: Почему если я пытаюсь сделать return из этого метода (return 0 или return "not_correct")  после throw "not_correct"; возникает необработаннное исключение. А если выражение корректное, return postfix; работает нормально.

alvls commented:
> Вопрос: Почему если я пытаюсь сделать return из этого метода (return 0 или return "not_correct")  после throw "not_correct"; возникает необработаннное исключение. А если выражение корректное, return postfix; работает нормально.
Дело не в return.
1. Любая функция, которая бросает исключение, должна быть вызвана (с учетом любой возможной глубины стека вызовов) из блока try.
2. Если этого не произошло, механизм поиска обработчика не срабатывает и возникает необработанное исключение.

alvls commented:
Работа   принята |  
-- | --
__Ваш текущий балл:__ |  
Функциональность: | 2,5
Качество проектирования: | 2,5
Качество   кода: | 2,5
Качество тестов: | 2,5
Бонус: | 0,4
Штраф: | 0,0
__Итого:__ | 10,4
